src/cotask_context.cpp

```diff
@@ -1,92 +1,108 @@
+#include <tbb/tbb.h>
+#include <boost/asio/executor_work_guard.hpp>
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/steady_timer.hpp>
 #include <cotask/cotask_context.hpp>
 #include <cotask/cotask_exception.hpp>
 
 namespace cotask {
 
-struct cotask_context_impl {
-    std::atomic<context_status> status = context_status::stopped;
-    boost::asio::io_context io;
-    tbb::task_arena arena;
-    tbb::task_group group;
+struct cotask_context_impl : public std::enable_shared_from_this<cotask_context_impl> {
+  std::atomic<context_status> status = context_status::stopped;
+  tbb::task_arena arena;
+  tbb::task_group group;
+  boost::asio::io_context io;
+  boost::asio::executor_work_guard<boost::asio::io_context::executor_type>
+      work_guard;
 
-    void start() {}
-    void stop() {}
+  void start() {}
+  void stop() {}
+
+  void attach(operation_context& op) {
+    if (op.timer()) {
+        detach(op);
+    }
+
+    op.timer() = std::make_unique<boost::asio::steady_timer>(io);
+    op.timer()->expires_after(op.interval());
+
+    auto wp = std::weak_ptr<cotask_context_impl>(shared_from_this());
+    op.timer()->async_wait([op, wp](const auto ec){
```

Comment:
Make timer a member of the cotask context  (hashmap of timer per operation context).  This keeps the operation context agnostic to the asynchronous and task based api behind the scenes.